airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/FileSystemConfigPersistence.java

```diff
@@ -75,7 +76,7 @@ public static ConfigPersistence createWithValidation(final Path storageRoot) thr
 
   public FileSystemConfigPersistence(final Path storageRoot) {
     this.storageRoot = storageRoot;
-    this.configRoot = storageRoot.resolve(CONFIG_DIR);
+    this.configRoot = Exceptions.toRuntime(() -> Files.createDirectories(storageRoot.resolve(CONFIG_DIR)));
```

Comment:
@cgardens, this is only needed for testing, right?

In production the config directory should not be created by this constructor. This is because when the server launches, it is possible that the config volume is not ready yet. It should wait for the config volume, instead of creating this directory itself. We have this logic in the [`createWithValidation` method](https://github.com/airbytehq/airbyte/blob/master/airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/FileSystemConfigPersistence.java#L62).

This is the PR with more details: #4835.

Reply:
ah. make sense. i'll revert this part out.

Reply:
fixed it here in this PR: https://github.com/airbytehq/airbyte/pull/4987